src/LandIce/evaluators/LandIce_ThicknessResid_Def.hpp

```diff
@@ -231,8 +233,13 @@ evaluateFields(typename Traits::EvalData workset)
         dH_Cell(node) = dH(elem_LID, node);
         H0_Cell(node) = H0(elem_LID, node);
         SMB_Cell(node) = have_SMB ? SMB(elem_LID, node) : ScalarT(0.0);
-        for (std::size_t dim = 0; dim < numVecFODims; ++dim)
-          V_Cell(node, dim) = V(elem_LID, elem_side, i, dim);
+        for (std::size_t dim = 0; dim < numVecFODims; ++dim) {
+          if (useCollapsedSidesets) {
+            V_Cell(node, dim) = V(iSide, i, dim);
```

Comment:
This comment isn't specifically related to this line or even this PR but it is the first time I've taken a look at this evaluator and I noticed something that may cause issues in the future.

It looks like for every single value of iSide, a number of dynamic rank views are being allocated. This is fine for CPU I imagine, but this will need to be changed eventually for GPU since allocating many small arrays on the device becomes much more expensive. Again, it isn't relevant for this PR but it might be worth adding a TODO comment.

Reply:
Yes, I think you're right. BTW, I think this was somewhat copied from what happens in Neumann evaluator. Except in Neumann, the temporaries are created in postRegSetup, with a size large enough to accommodate all sides, and during evaluation, they are subviewed at the proper offset location. Maybe the same mod can be done here too. Or maybe we should just go with raw Kokkos views (I don't understand the need for DynRankView).